FCM 설정 및 Tuist 마이그레이션

ray3238 opened the pull request:
## 개요
- FCM 설정
* +) 파베 버전 업그레이드로 인한 Tuist 실행 불가 이슈로 긴급 마이그레이션


## 작업사항
- FCM 설정
- Tuist 3.15.0 -> Tuist 4.23.0
- - Dependencies.swift -> Package.swift
- - .init() -> .target, .scheme, .project ...etc
- Google Service Info Run Script 설정
- Tests 타겟 info(default) -> info.plist(App/Tests/Support/Tests-Info.plist)
- Shared.xcconfig
- - iOS/watchOS minimum deployment target(iOS 15.0 / watchOS 9.0)
- - CFBundleVersion[CURRENT_PROJECT_VERSION] -> Shared.xcconfig에서 지정
- - Bundle version string(short)[CFBundleShortVersionString] -> Shared.xcconfig에서 지정
- DMS-XCConfig
- - Firebase Google Service Info.plist


## UI
![IMG_9379](https://github.com/user-attachments/assets/6ec701c7-f278-4d79-b645-a52e536fa2a2)

ray3238 committed:
ci :: GoogleServiceInfo 파일 추가

ray3238 committed:
feat :: 파이어베이스 FCM 설정

ray3238 committed:
feat :: 로그인API DTO 수정

baekteun commented on Projects/App/Sources/Application/AppDelegate.swift:
```diff
@@ -74,3 +95,40 @@ private extension AppDelegate {
+        didReceiveRegistrationToken fcmToken: String?
+    ) {
+        let token = String(describing: fcmToken)
+        print("Firebase registration token: \(token)")
```
print 지우는게 좋아보여요

baekteun commented on Projects/App/Sources/Application/AppDelegate.swift:
```diff
@@ -74,3 +95,40 @@ private extension AppDelegate {
+        _ application: UIApplication,
+        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+    ) {
+        print("APNS token: \(deviceToken)")
```
print 지우는게 좋아보여요

baekteun commented on Plugin/DependencyPlugin/ProjectDescriptionHelpers/Dependency+SPM.swift:
```diff
@@ -12,4 +12,9 @@ public extension TargetDependency.SPM {
+}
+
+public extension Package {
+    static let FirebaseMessaging = Package.remote(url: "https://github.com/firebase/firebase-ios-sdk", requirement: .upToNextMajor(from: "9.5.0"))
```
버전이 너무 낮은거같은데 9.5.0이 맞나요? 최근 10.x대 업데이트에 privacy manifest 추가하는 버전도 나왔어가지고 9.5.0으로 괜찮나요?

ray3238 replied:
다른 프로젝트에서 push 알림 기능만 사용하는데에는 9.5.0 버전으로도 괜찮았어서 사용했습니다.
버전 업그레이드가 되면서 privacy manifest라는 기능이 생겼는지는 몰랐네요. 조금 더 공부해보겠습니다.

DSMInhyeKang replied:
@ray3238 심사 올리기 전에 한 번 확인하고 올리는 게 나을 것 같아요.
https://github.com/firebase/firebase-ios-sdk/issues/11490

ray3238 committed:
refactor :: print문 삭제

ray3238 committed:
perf :: firebase 버전 업그레이드(11.0.0)

baekteun reviewed (approved):
CI는 언제 고쳐지는지 눈물나네요

DSMInhyeKang reviewed (changes requested):
페메로 말한 거 확인되면 다시 알려주세요

ray3238 committed:
refactor :: fcm 코드 코멘트 반영

DSMInhyeKang committed:
fix :: 일단 Tuist 4.23.0 업그레이드

DSMInhyeKang committed:
fix :: Dependencies.swift to Package.swift

error: 'tuist': invalid manifest (evaluation failed)
Invalid semantic version string '7.0'

DSMInhyeKang committed:
fix :: Tuist migration, CFBundleVersion(String), GoogleService Script

DSMInhyeKang committed:
ci :: ci_install_tuist 삭제 및 ci_post_clone mise 설치

DSMInhyeKang committed:
ci :: mise x 변경

DSMInhyeKang committed:
ci :: mise x to mise exec --

DSMInhyeKang committed:
ci :: PATH 삭제 및 mise install tuist

DSMInhyeKang committed:
ci : mise install.sh -> run

DSMInhyeKang committed:
ci :: pwd

DSMInhyeKang committed:
ci :: cd .. 제거

DSMInhyeKang committed:
ci :: 디렉토리 이동

DSMInhyeKang committed:
ci :: tuist install 전에 디렉토리 이동

DSMInhyeKang committed:
ci :: cd 시점 변경 및 mise install 사용

DSMInhyeKang committed:
ci :: 경로 및 파일 내용 확인

DSMInhyeKang committed:
ci :: rm -rf XCConfig

DSMInhyeKang committed:
chore :: mise swiftlint

DSMInhyeKang committed:
chore :: chmod +x ci_post_clone.sh, change SwiftLint Config Path

DSMInhyeKang committed:
chore :: SwiftLint Config Path 변경

DSMInhyeKang committed:
ci :: Google Service Info 파일 클론

DSMInhyeKang committed:
ci :: 디렉토리 이동 코드 제거 및 swiftlint config path 수정

DSMInhyeKang committed:
chore :: echo SRCROOT

DSMInhyeKang committed:
chore :: SwiftLintRunScript 복구

DSMInhyeKang committed:
chore :: .mise.toml swiftlint 삭제
